src/box/lua/iproto.c

```diff
@@ -240,6 +240,29 @@ lbox_iproto_session_new(struct lua_State *L)
 	return 1;
 }
 
+/**
+ * Drop all current connections.
+ *
+ * Accepts a timeout (in seconds).
+ *
+ * Returns nothing on success, raises an error on a failure.
+ *
+ * See iproto_drop_connections() for details.
+ */
+static int
+lbox_iproto_drop_connections(struct lua_State *L)
+{
+	int n_args = lua_gettop(L);
+	if (n_args != 1 || lua_type(L, 1) != LUA_TNUMBER)
+		return luaL_error(L, "Usage: box.iproto.internal."
+				  "drop_connections(timeout)");
+	double timeout = lua_tonumber(L, 1);
+	int rc = iproto_drop_connections(timeout);
+	if (rc < 0)
+		return luaT_error(L);
+	return 0;
+}
+
```

Comment:
Rough idea to be implemented later: use some (not sure what) default value, since in general the timeout is not so important. 

At the same time, this internal handle seems useful for tarantool/tarantool-ee#971

Reply:
TBH, I don't understand why it may take some visible time. A couple of shutdown+close calls, plus inter-thread messages. No synchronous network transmissions, I guess.

So maybe it shouldn't report any error, shouldn't have a timeout and if something goes bad, should just log it.

I'm not ready to make this decision now, because I didn't investigate/test it against these questions.